packages/less/plugin/compile-less.js

```diff
@@ -45,49 +44,66 @@ class LessCompiler extends MultiFileCachingCompiler {
              /\.lessimport$/.test(pathInPackage));
   }
 
+  compileOneFileLater(inputFile, getResult) {
+    inputFile.addStylesheet({
+      path: inputFile.getPathInPackage(),
+    }, async () => {
+      const result = await getResult();
+      return result && {
+        data: result.css,
+        sourceMap: result.sourceMap,
+      };
+    });
```

Comment:
If you're already using `CachingCompiler` or `MultiFileCachingCompiler` to implement your compiler plugins, you should implement a new method called `compileOneFileLater` that works roughly like this.